bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/discovery/HomematicDeviceDiscoveryService.java

```diff
@@ -109,15 +109,17 @@ public int getScanTimeout() {
     public synchronized void stopScan() {
         logger.debug("Stopping Homematic discovery scan");
         disableInstallMode();
-        thingHandler.getGateway().cancelLoadAllDeviceMetadata();
+        if (thingHandler.getGateway() != null) {
+            thingHandler.getGateway().cancelLoadAllDeviceMetadata();
+        }
```

Comment:
I'd suggest using variables:
```
final HomematicGateway gateway = thingHandler.getGetway();
if (gateway != null) {
    gateway.cancelLoadAllDeviceMetadata();
}
```

Same applies above and below.

Reply:
Should not be needed as the block is marked `synchronized`.

Reply:
fixed.